packages/commonwealth/client/scripts/hooks/useWallets.tsx

```diff
@@ -583,6 +620,13 @@ const useWallets = (walletProps: IuseWalletProps) => {
     } else {
       const selectedAddress = getAddressFromWallet(wallet);
 
+      // check if address exists
+      const addressExists = await doesAddressExist(wallet, selectedAddress);
```

Comment:
ATM, we can't use react query here for fetching, as this file is super fragile and there are a lot of race conditions in this file which break very easily.